libraries/AP_HAL_ChibiOS/UARTDriver.cpp

```diff
@@ -120,7 +120,7 @@ void UARTDriver::uart_thread()
         // change the thread priority if requested - if unbuffered it should only have higher priority than the owner so that
         // handoff occurs immediately
         if (mask & EVT_TRANSMIT_UNBUFFERED) {
-            chThdSetPriority(unbuffered_writes ? MIN(_uart_owner_thd->prio + 1, APM_UART_UNBUFFERED_PRIORITY) : APM_UART_PRIORITY);
+            chThdSetPriority(unbuffered_writes ? MIN(_uart_owner_thd->realprio + 1, APM_UART_UNBUFFERED_PRIORITY) : APM_UART_PRIORITY);
```

Comment:
So, `prio` could changes during runtime. Is this an issue in previous releases as well? Maybe backport is required if that's the case.

Reply:
This was my change, I think we should be picking the configured priority rather than the transient priority here, although they will largely be the same.